### DISCUSSION TITLE
Opaque type の日本語訳

### DISCUSSION THREAD

**Discussion by tsubasakat** [stzn/the-swift-programming-language-jp#321] [General]
### 訳をどうするか

この話の結論をすぐに出すのは難しいと思うので、
issue ではなく discussion を作りました。
Discussion だと、あとで人が探しやすいというのもあります。

### 現状

WWDC の字幕ではまだ訳されていないと思います。
日本語字幕の中でそこだけ `opaque type` となっていました。

これに対する訳は他の言語コミュニティでも定まっていなさそうで、
一番定着している訳というのもわからなかったです。

他方、データベースの型について調べると
簡単に変わることはないのではと思う例がありました。

IBM と Oracle は自社製品に関するドキュメントの中で `不透明型` という訳を使っています。

**Comment by tsubasakat** [comment]
僕はいつか Apple が訳すまでの間だけでも `不透明型` とするのはどうかなと思っています。

**Reply by stzn** [reply]
僕個人としては、不透明型だとちょっと何を指しているのかがわからない印象がありますね😅opaqueはopaqueとしてみんな理解していると思っているので、やるとしてもopaque型ですかねー。protocolを契約、genericsを一般化とは訳さないのに近い感覚があります。
ただ、オペーク型...だとちょっと違和感ありますね。ここは他の方のご意見もお伺いしたいですね。

**Reply by tsubasakat** [reply]
> protocolを契約、genericsを一般化とは訳さないのに近い感覚があります。

これ分かります！
いやそうはならんだろですね。

> オペーク型...だとちょっと違和感ありますね。

不透明型でない時はプロトコルやジェネリクスに倣ってオペーク型はどうかなと思ったのですが… 💧

妥当な訳がない時、読む側のことを考えれば翻訳しすぎない方が一番 contribution かも知れないですね。
